tools/gather-sysinfo/gather-sysinfo.go

```diff
@@ -101,3 +83,37 @@ func chrootFileSpecs(fileSpecs []string, root string) []string {
 	}
 	return entries
 }
+
+func dedupExpectedContent(fileSpecs, extraFileSpecs []string) []string {
+	specSet := make(map[string]int)
+	for _, fileSpec := range fileSpecs {
+		specSet[fileSpec]++
+	}
+	for _, extraFileSpec := range extraFileSpecs {
+		specSet[extraFileSpec]++
+	}
+
+	var retSpecs []string
+	for retSpec := range specSet {
+		retSpecs = append(retSpecs, retSpec)
+	}
+	return retSpecs
+}
+
+func kniExpectedCloneContent() []string {
+	return []string{
+		// generic information
+		"/proc/cmdline",
```

Comment:
I see the list is much smaller than the original one. I assume you already "de-duplicated" offline the files?
Maybe we should consider listing here all the files we need and let the code do the de-duplication?
It will be like a documentation of the files we need, otherwise we will have the list in more than one place.
Only a thought.

Reply:
The list now contains only the files we (KNI) need on top of ghw's, that's why is much smaller.
because of net and PCI topologies, which are host-specific, it's unpractical to have a static list; the only feasible option is to create the list dinamically, and to do so - and to be able to inspect that list, we have the `dump` option.

The code which creates the list and manages the snapshot is vendored, so we can inspect it easily.

Reply:
re: net and PCI, we have unmerged PRs to address this, see https://github.com/jaypipes/ghw/pull/224

Reply:
got it, thanks